src/main/kotlin/org/rust/ide/intentions/ExtractInlineModule.kt

```diff
@@ -0,0 +1,45 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.actions.ReformatCodeProcessor
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RustElementFactory
+import org.rust.lang.core.psi.RustModItem
+import org.rust.lang.core.psi.util.isTopLevelInFile
+import org.rust.lang.core.psi.util.parentOfType
+
+class ExtractInlineModule : IntentionAction {
+    override fun getFamilyName() = "Extract inline module structure"
+    override fun getText() = "Extract Inline Module"
+    override fun startInWriteAction() = true
+
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?) {
+        file ?: return
+        val offset = editor?.caretModel?.offset ?: return
+        val element = file.findElementAt(offset)?.parentOfType<RustModItem>() ?: return
+        val elementName = element.name ?: return
+        val fileName = "$elementName.rs"
+        val modFile = file.parent?.createFile(fileName)
+
+        val copyElement = element.copy() as? RustModItem ?: return
+        copyElement.mod.delete()
+        copyElement.rbrace?.delete()
+        copyElement.lbrace?.delete()
+        copyElement.nameElement?.delete()
+
+        modFile?.add(copyElement)
```

Comment:
I think it's better to use `addRange` function here. Here is a [first example](https://upsource.jetbrains.com/idea-ce/file/idea-ce-5fec12b146609338bee5dee5208dd1a011a1fabd/java/java-analysis-impl/src/com/intellij/codeInsight/daemon/impl/quickfix/ChangeNewOperatorTypeFix.java?nav=5415:5632:focused&line=159) I've found, there may be a better one.